### PR TITLE
Added connection retrying in case of failure

### PR DESCRIPTION
**Pull request by Peterburnett**

### CODE REVIEW

**Comment by Peterburnett** [comment]
Hi @jonof  Are you able to review this patch at any point?

File: classes/client.php [hunk 1]
```diff
@@ -190,23 +190,42 @@ public static function is_error_result($result) {
     public function connect($type, $host) {
         $this->disconnect();
 
-        if (!$this->open_socket($type . '://' . $host, $errno, $errstr)) {
-            throw new moodle_exception('errorcantopen'.$type.'socket', 'antivirus_savdi', '', "$errstr ($errno)");
-        }
+        $connected = false;
+        $tries = (int) get_config('antivirus_savdi', 'connretry');
```
Comment (jonof) [inline classes/client.php]: Is there any particular reason why the retry count isn't passed into `connect()` as a parameter considering no other configuration is being read in by the _client_ class directly?

Comment (Peterburnett) [reply classes/client.php]: No particular reason, I was mostly mirroring the pattern from here for consistency:

https://github.com/moodle/moodle/blob/master/lib/antivirus/clamav/classes/scanner.php#L75

Happy to change it to a method parameter.

Comment (Peterburnett) [reply classes/client.php]: Also a function of me not looking at the wider context as I should have while juggling a few things. Fixed

**Pull request merged**